src/index/api.rs

```diff
@@ -168,22 +161,144 @@ impl Api {
         )?
         .json()?;
 
-        let result = response
+        Ok(response
             .users
             .into_iter()
-            .filter(|data| {
-                !data
-                    .login
-                    .as_ref()
-                    .map(|login| login.is_empty())
-                    .unwrap_or_default()
-            })
-            .map(|data| CrateOwner {
-                avatar: data.avatar.unwrap_or_default(),
-                login: data.login.unwrap_or_default(),
-            })
-            .collect();
-
-        Ok(result)
+            .filter(|data| !data.login.is_empty())
+            .collect())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use mockito::mock;
+    use serde_json::{self, json};
+
+    fn api() -> Api {
+        Api::new(Some(Url::parse(&mockito::server_url()).unwrap())).unwrap()
+    }
+
+    #[test]
+    fn get_owners() {
+        let _m = mock("GET", "/api/v1/crates/krate/owners")
+            .with_header("content-type", "application/json")
+            .with_body(
+                serde_json::to_string(&json!({
+                "users": [
+                    {
+                        "id": 1,
+                        "login": "the_first_owner",
+                        "name": "name",
+                        "avatar": "http://something"
+                    },
+                    {
+                        "id": 2,
+                        "login": "the_second_owner",
+                        "name": "another name",
+                        "avatar": "http://anotherthing"
+                    }
```

Comment:
Let's include all the fields from real API format here, please, so we can make sure we don't error on unknown fields.
```suggestion
                    {
                        "id": 1,
                        "login": "the_first_owner",
                        "name": "name",
                        "avatar": "http://something",
                        "kind": "user",
                        "url": "https://github.com/the_second_owner"
                    },
                    {
                        "id": 2,
                        "login": "the_second_owner",
                        "name": "another name",
                        "avatar": "http://anotherthing",
                        "kind": "user",
                        "url": "https://github.com/the_second_owner"
                    }
```

Reply:
added